dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh

```diff
@@ -279,16 +279,55 @@ echo "done!"
 echo -n "[CHE] Checking if project \"${CHE_OPENSHIFT_PROJECT}\" exists..."
 if ! oc get project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null; then
 
-  if [ "${COMMAND}" == "cleanup" ] || [ "${COMMAND}" == "rollupdate" ]; then echo "**ERROR** project doesn't exist. Aborting"; exit 1; fi
-  if [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then echo "**ERROR** project doesn't exist on OSIO. Aborting"; exit 1; fi
+    if [ "${COMMAND}" == "cleanup" ] || [ "${COMMAND}" == "rollupdate" ]; then echo "**ERROR** project doesn't exist. Aborting"; exit 1; fi
+    if [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then echo "**ERROR** project doesn't exist on OSIO. Aborting"; exit 1; fi
 
-  echo -n "Project does not exist...creating it..."
-  oc new-project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null
+    # OpenShift will not get project but project still exists for a period after being deleted.
+    # The following will loop until it can create successfully.
+
+    WAIT_FOR_PROJECT_TO_DELETE=true
+    WAIT_FOR_PROJECT_TO_DELETE_MESSAGE="Waiting for project to be deleted fully(~15 seconds)..."
+
+    echo "Project \"${CHE_OPENSHIFT_PROJECT}\" does not exist...trying to creating it."
+    DEPLOYMENT_TIMEOUT_SEC=120
+    POLLING_INTERVAL_SEC=2
+    timeout_in=$((POLLING_INTERVAL_SEC+DEPLOYMENT_TIMEOUT_SEC))  
+    while $WAIT_FOR_PROJECT_TO_DELETE
+    do
+    { # try
+        timeout_in=$((timeout_in-POLLING_INTERVAL_SEC))
+        if [ "$timeout_in" -le "0" ] ; then
+            echo "[CHE] **ERROR**: Timeout of $DEPLOYMENT_TIMEOUT_SEC waiting for project \"${CHE_OPENSHIFT_PROJECT}\" to be delete."
+            exit 1
+        fi  
+        oc new-project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null && \
+        WAIT_FOR_PROJECT_TO_DELETE=false # Only excutes if project creation is successfully
+    } || { # catch
+        echo -n $WAIT_FOR_PROJECT_TO_DELETE_MESSAGE
+        WAIT_FOR_PROJECT_TO_DELETE_MESSAGE="."
+        sleep $POLLING_INTERVAL_SEC
+    }
+    done
 fi
-echo "done!"
+echo "Project \"${CHE_OPENSHIFT_PROJECT}\" creation done!"
 
 echo -n "[CHE] Switching to \"${CHE_OPENSHIFT_PROJECT}\"..."
-oc project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null
+WAIT_TO_SWITCH_TO_PROJECT=true
```

Comment:
Do we really need this waiting? As far as I see a current project is automatically switched to newly created after `oc new-project`. When there can be a delay between project creation (`oc new-project`) and ability to switch to it.

Reply:
@sleshchenko I think the intent was to allow switch if project already existed but I cannot see the reason to do this anymore. Removed.